src/improvements/AddressRegistry.sol

```diff
@@ -73,38 +104,104 @@ contract AddressRegistry is IAddressRegistry, Test {
             chains.push(_chains[i]);
         }
 
-        /// should never revert
         string memory chainAddressesContent =
             vm.readFile("lib/superchain-registry/superchain/extra/addresses/addresses.json");
 
         for (uint256 i = 0; i < chains.length; i++) {
-            uint256 chainId = chains[i].chainId;
-            string memory chainName = chains[i].name;
-            require(!supportedL2ChainIds[chainId], "Duplicate chain ID in chain config");
-            require(chainId != 0, "Invalid chain ID in config");
-            require(bytes(chainName).length > 0, "Empty name in config");
+            _processChain(chains[i], chainAddressesContent);
+        }
+    }
+
+    /// @dev Processes all configuration for a single chain.
+    function _processChain(ChainInfo memory chain, string memory chainAddressesContent) internal {
+        uint256 chainId = chain.chainId; // L2 chain ID.
 
-            supportedL2ChainIds[chainId] = true;
+        require(!supportedL2ChainIds[chainId], "Duplicate chain ID in chain config");
+        require(chainId != 0, "Invalid chain ID in config");
+        require(bytes(chain.name).length > 0, "Empty name in config");
 
-            string[] memory keys = vm.parseJsonKeys(chainAddressesContent, string.concat("$.", vm.toString(chainId)));
+        supportedL2ChainIds[chainId] = true;
 
-            for (uint256 j = 0; j < keys.length; j++) {
-                string memory key = keys[j];
-                address addr =
-                    vm.parseJsonAddress(chainAddressesContent, string.concat("$.", vm.toString(chainId), ".", key));
+        address optimismPortalProxy = _findOptimismPortalAndSaveEntries(chain, chainAddressesContent);
 
-                require(addr != address(0), "Invalid address: cannot be zero");
-                require(
-                    registry[key][chainId].addr == address(0),
-                    "Address already registered with this identifier and chain ID"
-                );
+        address superchainConfig = getSuperchainConfig(optimismPortalProxy);
+        if (superchainConfig != address(0)) {
```

Comment:
Were there some chains that didn't have a superchain config? If so, whew that is an old version and I think we can ignore that chain to simplify code. Any chain we hold keys for is on at least `op-contracts/v1.8.0` which has a superchain config

Reply:
https://github.com/ethereum-optimism/superchain-ops/pull/565/commits/d8d4b196c692eac2c67a859d1bb3e5cfca940ee7